lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp

```diff
@@ -164,6 +164,212 @@ int send_uplink_nas_registration_complete(
   return (rc);
 }
 
+/* Create pdu session establishment  request from ue */
+int send_uplink_nas_pdu_session_establishment_request(
+    amf_app_desc_t* amf_app_desc_p, amf_ue_ngap_id_t ue_id, const plmn_t& plmn,
+    const uint8_t* nas_msg, uint8_t nas_msg_length) {
+  bstring pdu_session_est_req;
+  tai_t originating_tai = {};
+
+  if ((!amf_app_desc_p) || (!nas_msg) || (nas_msg_length == 0)) {
+    return RETURNerror;
+  }
+
+  pdu_session_est_req = blk2bstr(nas_msg, nas_msg_length);
+  memcpy(pdu_session_est_req->data, nas_msg, nas_msg_length);
+
+  originating_tai.plmn = plmn;
+  originating_tai.tac  = 1;
+
+  int rc = RETURNerror;
+  rc     = amf_app_handle_uplink_nas_message(
+      amf_app_desc_p, pdu_session_est_req, ue_id, originating_tai);
+
+  return rc;
+}
+
+void create_ip_address_response_itti(
+    itti_amf_ip_allocation_response_t& response) {
+  std::string apn = "internet";
+  std::copy(apn.begin(), apn.end(), std::begin(response.apn));
+  response.default_ambr.br_unit    = BPS;
+  response.default_ambr.br_dl      = 200000000;
+  response.default_ambr.br_ul      = 100000000;
+  response.gnb_gtp_teid            = 0;
+  response.gnb_gtp_teid_ip_addr[0] = 0xc0;
+  response.gnb_gtp_teid_ip_addr[1] = 0xa8;
+  response.gnb_gtp_teid_ip_addr[2] = 0x3c;
+  response.gnb_gtp_teid_ip_addr[3] = 0x96;
+  std::string imsi                 = "222456000000001";
+  std::copy(imsi.begin(), imsi.end(), std::begin(response.imsi));
+  response.imsi_length  = 15;
+  response.paa.pdn_type = IPv4;
+  inet_pton(AF_INET, "192.168.128.254", &(response.paa.ipv4_address));
+  response.paa.vlan         = 1;
+  response.pdu_session_id   = 1;
+  response.pdu_session_type = IPv4;
+  response.pti              = 0x01;
+  response.result           = 0;
```

Comment:
Can we use macros instead of hardcoded numeric values?

Reply:
the above value will change for each unit test case, these values are to compare Hex buffer received UE, which will change each registration/PDU.

I agree with your point if any hard-coded values in the AMF application code, I will take this comment as a note.